Shears can defuse TNT

Findus642 opened the pull request:
TNT can now be defused when right clicking with shears, nothing more to it than an additional gameplay mechanic. 

Option is per-world and a value of 0.00 disables the function. See patch commit for a few more details

Findus642 committed:
Shears can defuse TNT

Shears can now defuse TNT when right clicked

granny commented on patches/server/0306-Shears-can-defuse-TNT.patch:
```diff
@@ -0,0 +1,77 @@
+ import net.minecraft.world.item.DyeColor;
+ import net.minecraft.world.item.Item;
+ import net.minecraft.world.item.Items;
+-import net.minecraft.world.level.Explosion;
```
needs to be reverted

granny commented on patches/server/0306-Shears-can-defuse-TNT.patch:
```diff
@@ -0,0 +1,77 @@
+ import java.util.ArrayList;
+ import java.util.HashMap;
+ import java.util.List;
+-import java.util.Locale;
```
needs to be reverted.

granny reviewed (changes requested):
Currently on mobile so i can't look into the implementation, so nitpicking for now until I can take a better look. Patch diffs should touch as little as possible to lower the maintenance burden of each patch.

Findus642 committed:
Revert unused imports

MrFishCakes commented:
Both been reverted. Didn't pick up on this during committing!

Findus642 committed:
Patch changes for optimisation(s) and safety checks

Checks if level is client sided or not, adds permission for defusing TNT (purpur.tnt.defuse), handles damaging item more appropriately and item drop to mimic vanilla behaviour better. Formatted code slightly better incase future implementations are used within the method.

granny reviewed (approved):
LGTM :)

All that's left is to document the config option and permission in the docs (https://github.com/purpurMC/purpurdocs) and the patch will be merged!
